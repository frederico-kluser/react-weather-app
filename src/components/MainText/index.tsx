import MainTextContainer, { H1, H2, H3 } from './styled';

interface MainTextInterface {
  temperature: string;
  status: string;
  lastUpdate: string;
}

const MainText = ({temperature, status, lastUpdate}: MainTextInterface) => (
  <MainTextContainer>
    <H1>{temperature}Â°C</H1>
    <H2>{status}</H2>
    <H3>Updated {lastUpdate}</H3>
  </MainTextContainer>
)

export default MainText;
