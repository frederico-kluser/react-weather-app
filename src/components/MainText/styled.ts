import styled from 'styled-components';
import { mobileChanges } from '../../utils/style';

const margin = 'margin: 0px;';
const marginTop = 'margin-top: 25px;';

const MainTextContainer = styled.div`
  text-align: center;
`;

export const H1 = styled.h1`
  font-size: 9em;
  margin-top: 0;
  ${margin}
  ${mobileChanges(`
    font-size: 6em;
    ${marginTop}
  `)}
`;

export const H2 = styled.h2`
  font-size: 2.4em;
  font-weight: 100;
  ${margin}
  margin-top: -25px;
  text-transform: capitalize;
  ${mobileChanges(marginTop)}
`;

export const H3 = styled.h3`
  font-size: 1.26em;
  font-weight: 100;
  margin-top: 10px;
  ${margin}
  ${mobileChanges(marginTop)}
`;


export default MainTextContainer;
