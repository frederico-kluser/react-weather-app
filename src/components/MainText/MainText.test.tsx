import renderer from 'react-test-renderer';
import MainText from '.';

test('MainText Renderer', () => {
  const component = renderer.create(
    <MainText temperature={20} status="Heavy shower snow" lastUpdate="14:40" />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});