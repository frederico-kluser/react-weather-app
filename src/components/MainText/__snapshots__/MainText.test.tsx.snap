// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainText Renderer 1`] = `
<div
  className="sc-bdvvtL bKhKNO"
>
  <h1
    className="sc-gsDKAQ bAFsMA"
  >
    20
    Â°C
  </h1>
  <h2
    className="sc-dkPtRN eBfRnW"
  >
    Heavy shower snow
  </h2>
  <h3
    className="sc-hKwDye dzNrnq"
  >
    Atualizado 
    14:40
  </h3>
</div>
`;
